### ISSUE TITLE
"Find In Files progress..." window dialog box is hidden behind Notepad++ window and the main Notepad++ window do not respond

### ISSUE THREAD

**Issue by Neustradamus** [notepad-plus-plus/notepad-plus-plus#13972]
### Description of the Issue
When we do a search in Files, the "Find In Files progress..." window dialog box is hidden behind Notepad++ window and the main Notepad++ window and "Find in Files" window do not respond ("Find In Files progress..." window works)

### Steps to Reproduce the Issue
1. Open Notepad++
2. "Search" menu -> "Find in files" CTRL+SHIFT+F
3. Select a directory
4. Enter "Find what"
5. "Find All" button
6. Open another software
7. Click in Notepad++ and the "Find In Files progress..." window dialog box is hidden behind Notepad++ window.

### Expected Behavior
To see the "Find In Files progress..." window dialog box
To see Notepad++ responds normally

### Actual Behavior
Main Notepad++ window and "Find in Files" window do not respond
"Find In Files progress..." window works but hidden behind Notepad++

**Comment by Neustradamus** [comment]
@notepad-plus-plus, @donho: Have you tried?
It is an historical bug.

**Comment by forderud** [comment]
I'm also experiencing the same problem of the "Find In Files progress..." window dropping behind the "Find in Files" window and the rest of Notepad++. This is rather annoying.

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/2671400/16b14dbe-4da8-4bf4-becd-9749dcedc862)

Investigation of the Windows in Spy++ reveal that the problem is most likely caused by lack of parent/child relationship between the windows:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/2671400/81763977-61ee-476e-bb27-30e2e551c666)

**Comment by xomx** [comment]
I can confirm the problem.

Did you notice this (it is not a simple z-order problem):

![npp_progresswnd_problem_-_separate_taskbar_wnd](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/13075183/f32ba05d-74b3-4d05-9668-cf696588e9a9)

Why it behaves like that (using `WS_EX_APPWINDOW` for the `dwExStyle` and `NULL` as the `hWndParent`):
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/8e43b0adbaed0353a7a97c2ff4e2da3e2dc8efa8/PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp#L5687

I will see what I can do (no promises as it probably will not be one-src-line-change...).

**Comment by forderud** [comment]
> Why it behaves like that (using `WS_EX_APPWINDOW` for the `dwExStyle` and `NULL` as the `hWndParent`):
> ...

Agree. You'll probably need to pass the "Find in Files" window handle as `hWndParent` when creating the "Find In Files progress..." window to ensure that it's always drawn in front of the other window.

> I will see what I can do (no promises as it probably will not be one-src-line-change...).

That would be great. A big thanks in advance.

**Comment by xomx** [comment]
@Yaron10 

> 1. Searching for `a` under `C:\` (in all sub-folders), I get a black screen, "Not responding" and NPP may even crash.

"Not responding" is expected in such case, but not the crash (at least for me it never crashed, I only got looong delays before the progress-wnd showed up - the more files I had on the "C:" drive, the longer delays I have got). This is caused by the current N++ 2-stage design:

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/97dd708e233559a4a0bd819f7ee72892c27e1a66/PowerEditor/src/Notepad_plus.cpp#L2135-L2142

1. In the `createFilelistForFiles(fileNames)` N++ creates beforehand list of all the absolute filenames that need to be searched - this is probably your "hang/crash" N++ state (the delay also depends on the speed of your disk, a fast SSD helps here...)
2. Then in the `findInFilelist(fileNames)` the real searching is done, this time with the progress-wnd showing the currently processed file

It is (to put it mildly) very sub-optimal behavior.

I would like to change it to something like:
- not using a separate taskbar progress window at all, so user always gets the responding progress-wnd at the top of the N++ (non-responding) one
- at the 1st stage also showing the progress-wnd, just with a rotating "busy" indicator (doable e.g. by firing up a WM_TIMER and updating the progress-wnd in some intervals)
- at the 2nd stage show the progress filenames as usual


> Would the result be different if the Progress-window was removed/neutralized? - Probably not.

In such case the delay/hang will be longer (as it will cover both of the above stages...).

> 2\. Can you reproduce the "Progress-window hidden behind..." issue in a relatively short search where you don't get "Not responding"?

Try to search fewer files, e.g. use "C:\Windows\" instead of the whole "C:\".

To bring the hidden N++ progress-wnd to the front, I use the [System Informer](https://systeminformer.sourceforge.io/) (formerly Process Hacker):

![npp-progresswnd-SystemInformer](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/13075183/473d0999-8568-40bf-b9f3-d0fd3957e391)

**Comment by Yaron10** [comment]
@xomx,

Thank you for looking into it and for the detailed analysis.

Let's put aside the performance/crash issue.
(Notepad++ - **great as it is** - has quite a few major performance issues).

I simply can not reproduce the "Progress-window hidden behind..." issue.
(The Find dialog title is : _Find in Files (Not responding)_).

![תמונה](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/10229320/6be52024-e577-48e7-bae7-f02158f93a30)

***

System Informer is brilliant (thank you @dmex!).

**Comment by Neustradamus** [comment]
Thanks all for your comments :)

At the same time, the icon of "Find In Files"window is missing (you can see in different screenshots).

**Comment by Yaron10** [comment]
@Neustradamus,

> At the same time, the icon of "Find In Files"window is missing

👍 
That's a nice observation.
Normally , when we don't get "Not responding", that image (or any icon) is not displayed.

**Comment by Yaron10** [comment]
@Neustradamus,

Can you reproduce the issue in the following cases?

1. On a executing a relatively short search where you don't get "Not Responding"?
2. On using 96 DPI system-wide (scaling: 100%)?

**Comment by Neustradamus** [comment]
Any progress on it?

The dialog window box is always at bottom...

**Comment by xomx** [comment]
@Neustradamus 
If that bump is for me - sorry, no progress.

**Comment by Neustradamus** [comment]
@xomx :/

If someone can add the missing Notepad++ icon in window dialog boxes:
- Find in Files
- Find In Files progress...

**Comment by xomx** [comment]
@Neustradamus 
There seems to be someone with more free time and interest in the FindInFiles area, you could try to ask him:
https://community.notepad-plus-plus.org/topic/25452/looking-for-issues-to-contribute

**Comment by Neustradamus** [comment]
@xomx and others: Have you progressed on it?

**Comment by xomx** [comment]
@Neustradamus 
I cannot reproduce this issue anymore in the current version.
Seems to me that the v8.6.8 also solved this problem somehow(?).
Could you confirm?

**Comment by Neustradamus** [comment]
@donho: Any progress?

It is possible to replace the bad icon to good one for "Find in Files" window?
And to add the icon for "Find In Files progress" window which is always missing?

Thanks in advance.

**Comment by xomx** [comment]
> I cannot reproduce this issue anymore in the current version.
> Seems to me that the v8.6.8 also solved this problem somehow(?).

So I found the answer here: https://community.notepad-plus-plus.org/post/103337

So this additional line changed the behavior:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/1fbff24482434610f9b0ec082424470c47ed5cc5/PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp#L6658

**Comment by xomx** [comment]
> It is possible to replace the bad icon to good one for "Find in Files" window?

Yes, it is possible. That "bad-icon" is probably the default system icon forced to us by WinOS dialog manager when we click on the inactive user-input blocked FindInFiles dlg (that clicking makes it behave like a top-level overlapped wnd with a caption icon after the rejected dlg-activation attempt...).

The remedy is easy - we can supply such a small caption icon ourselves here

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/9b558895e744601842ffdaff4679467863b9722d/PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp#L302-L304

by adding this oneliner after the above `StaticDialog::create`

`::SendMessageW(_hSelf, WM_SETICON, ICON_SMALL, (LPARAM)::LoadIcon(_hInst, MAKEINTRESOURCE(IDI_M30ICON)));`

@donho 
The FindInFiles with my above simple fix:
<img width="646" height="171" alt="Image" src="https://github.com/user-attachments/assets/d3385fcb-9f05-41a8-8d6a-84ff7a822eb2" />

Do you like that icon in FindInFiles or not? (it will be always displayed, not only in that inactive "not responding" state)
